### PR TITLE
Applied transition to specific Button properties

### PR DESCRIPTION
**Pull request by dimitrisnl**
These properties are:
- Color
- Background
- Background-color
- Border-color
- Box-shadow

Fixes: #3145
Signed-off-by: Dimitrios Lytras <dnlytras@gmail.com>

#### What does this PR do?

The PR excludes properties like `border-radius` from being 'animated'. Only color properties should have a transition effect.

#### What are the relevant issues?

#3145 

#### Do the grommet docs need to be updated?

No

#### Release notes
from @ericsoderberghp I think we should remove background. And, I think we need to highlight this change in the release notes to mention that people who have done extensive style theming might need to add some transition styles.

#### Is this change backwards compatible or is it a breaking change?

It is backward compatible

### CODE REVIEW

**Comment by oorestisime** [comment]
@ShimiSun @ericsoderberghp would love your input for this as well! LGTM

File: src/js/components/Button/StyledButton.js [hunk 1]
```diff
@@ -127,7 +127,13 @@ const StyledButton = styled.button`
   ${props =>
     !props.plain &&
     `
-    transition: 0.1s ease-in-out;
+    transition-property: color,
+      background,
```
Comment (ericsoderberghp) [inline src/js/components/Button/StyledButton.js]: I'm wondering why we need `background`. I am thinking that `background-color` might be sufficient?

**Comment by ericsoderberghp** [comment]
Depending on what caller's have done in their themes, this has an outside chance of being a breaking change. For instance, if the caller used `extend` with `:hover` in their theme and customized the padding, This change could remove the transition. I realize this is unlikely. I also think it would be fairly easy to rectify by updating the custom theme to include `transition-property` in their `extend`. I think we should include a note in the next release notes pointing this out, just in case.

**Comment by dimitrisnl** [comment]
@ShimiSun 	@ericsoderberghp Any updates on this? Should I remove the `background` property?

**Comment by ericsoderberghp** [comment]
> @ShimiSun @ericsoderberghp Any updates on this? Should I remove the `background` property?

I think we should. And, I think we need to highlight this change in the release notes to mention that people who have done extensive style theming might need to add some transition styles.

**Comment by dimitrisnl** [comment]
@ericsoderberghp 	Is there something I need to do from my end about the release notes? Are they populated from commit description/tags etc?

**Comment by ericsoderberghp** [comment]
The release notes are manually built using the commit description as reference material. So, we would just want to make sure the description is rich enough to be use to write the release notes.

**Comment by ShimiSun** [comment]
@dimitrisnl will you be willing to remove the `background`? I'll take care of the release notes.

**Comment by dimitrisnl** [comment]
@ShimiSun 	Yes, ofc. I have already removed this property.

**Comment by oorestisime** [comment]
Thanks for your patience! I updated the failing jest tests and getting this in! :tada: :+1:

**Pull request merged**